pal/AzureIoTSocket_Ethernet/socketio_esp32ethernet2.cpp

```diff
@@ -0,0 +1,577 @@
+                result = MU_FAILURE;
+            }
+        }
+    }
+
+    if (addrInfoIp != NULL)
+    {
+        freeaddrinfo(addrInfoIp);
+    }
+
+    return result;
+}
+CONCRETE_IO_HANDLE socketio_create(void* io_create_parameters)
+{
+    SOCKETIO_CONFIG* socket_io_config = (SOCKETIO_CONFIG*)io_create_parameters;
+    SOCKET_IO_INSTANCE* result;
+
+    if (socket_io_config == NULL)
+    {
+        LogError("Invalid argument: socket_io_config is NULL");
+        result = NULL;
+    }
+    else
+    {
+        result = (SOCKET_IO_INSTANCE*)malloc(sizeof(SOCKET_IO_INSTANCE));
+        if (result != NULL)
+        {
+            result->pending_io_list = singlylinkedlist_create();
+            if (result->pending_io_list == NULL)
+            {
+                LogError("Failure: singlylinkedlist_create unable to create pending list.");
+                free(result);
+                result = NULL;
+            }
+            else
+            {
+                if (socket_io_config->hostname != NULL)
+                {
+                    result->hostname = (char*)malloc(strlen(socket_io_config->hostname) + 1);
+                    if (result->hostname != NULL)
+                    {
+                        (void)strcpy(result->hostname, socket_io_config->hostname);
+                    }
+
+                    result->socket = NULL;
+                }
+                else
+                {
+                    result->hostname = NULL;
+                    *result->socket = *((EthernetClient*)socket_io_config->accepted_socket);
+                }
+
+                if ((result->hostname == NULL) && (result->socket == NULL))
+                {
+                    LogError("Failure: hostname == NULL and socket is invalid.");
+                    singlylinkedlist_destroy(result->pending_io_list);
+                    free(result);
+                    result = NULL;
+                }
+                else
+                {
+                    result->port = socket_io_config->port;
+                    result->target_mac_address = NULL;
+                    result->on_bytes_received = NULL;
+                    result->on_io_error = NULL;
+                    result->on_bytes_received_context = NULL;
+                    result->on_io_error_context = NULL;
+                    result->io_state = IO_STATE_CLOSED;
+                }
+            }
+        }
+        else
+        {
+            LogError("Allocation Failure: SOCKET_IO_INSTANCE");
+        }
+    }
+
+    return result;
+}
+
+void socketio_destroy(CONCRETE_IO_HANDLE socket_io)
+{
+    if (socket_io != NULL)
+    {
+        SOCKET_IO_INSTANCE* socket_io_instance = (SOCKET_IO_INSTANCE*)socket_io;
+        /* we cannot do much if the close fails, so just ignore the result */
+        if (socket_io_instance->socket != NULL)
+        {
+            if (socket_io_instance->hostname != NULL)
+            {
+                // We created this socket so delete it
```

Comment:
>// We created this socket so delete it [](start = 16, length = 38)

Can you please deallocate hostname ... for which you made null check.

Reply:
You have done that on line number 300, why are you checking for hostname and then doing delete on socket?


---
In reply to: [350497422](https://github.com/Azure/azure-iot-pal-arduino/pull/7#discussion_r350497422) [](ancestors = 350497422)

Reply:
Old code. Will update to delete hostname as well.

Reply:
Can you please check the latest?

---
In reply to: [350986016](https://github.com/Azure/azure-iot-pal-arduino/pull/7#discussion_r350986016) [](ancestors = 350986016)

Reply:
Yep, removed ethernet library.